aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -617,6 +627,67 @@ void bgemm_internal_cublas<at::Half>(CUDABLAS_BGEMM_ARGTYPES(at::Half)) {
 #endif // USE_ROCM
 }
 
+template <>
+void bgemm_internal_cublas<at::Half, float>(CUDABLAS_BGEMM_ARGTYPES_AND_C_DTYPE(at::Half, float)) {
```

Comment:
this looks like a copy paste from `<at::Half>` specialization with the only difference in c dtype, perhaps factor it out as a helper function and set c dtype with std::is_same
```
template <>
void bgemm_internal_cublas<at::Half, at::Half>(CUDABLAS_BGEMM_ARGTYPES(at::Half)) {
  bgemm_internal_cublas_helper<at::Half>(CUDABLAS_BGEMM_ARGS_AND_C_DTYPE(at::Half, at::Half));
}

template <>
void bgemm_internal_cublas<at::Half, float>(CUDABLAS_BGEMM_ARGTYPES_AND_C_DTYPE(at::Half, float)) {
  bgemm_internal_cublas_helper<float>(CUDABLAS_BGEMM_ARGS(at::Half));
}
```